docs/protocol/core/v3.0.rst

```diff
@@ -307,89 +333,111 @@ identify a data type.
 Core data types
 ---------------
 
+.. endianess disapear
+
 .. list-table:: Data types
    :header-rows: 1
 
    * - Identifier
      - Numerical type
      - Size (no. bytes)
      - Byte order
-   * - `bool`
-     - Boolean, with False encoded as `\\x00` and True encoded as `\\x01`
+   * - ``bool``
+     - Boolean, with False encoded as ``x00`` and True encoded as ``x01``
      - 1
      - None
-   * - `i1`
+   * - ``i1``
      - signed integer
      - 1
      - None
-   * - `<i2`
+   * - ``<i2``
      - signed integer
      - 2
      - little-endian
-   * - `<i4`
+   * - ``<i4``
      - signed integer
      - 4
      - little-endian
-   * - `<i8`
+   * - ``<i8``
      - signed integer
      - 8
      - little-endian
-   * - `>i2`
+   * - ``>i2``
      - signed integer
      - 2
      - big-endian
-   * - `>i4`
+   * - ``>i4``
      - signed integer
      - 4
      - big-endian
-   * - `>i8`
+   * - ``>i8``
      - signed integer
      - 8
      - big-endian
-   * - `u1`
+   * - ``u1``
      - signed integer
      - 1
      - None
-   * - `<u2`
+   * - ``<u2``
      - unsigned integer
      - 2
      - little-endian
-   * - `<u4`
+   * - ``<u4``
      - unsigned integer
      - 4
      - little-endian
-   * - `<u8`
+   * - ``<u8``
      - unsigned integer
      - 8
      - little-endian
-   * - `<f2`
+   * - ``<f2``
      - half precision float: sign bit, 5 bits exponent, 10 bits mantissa
      - 2
      - little-endian
-   * - `<f4`
+   * - ``<f4``
      - single precision float: sign bit, 8 bits exponent, 23 bits mantissa
      - 4
      - little-endian
-   * - `<f8`
+   * - ``<f8``
      - double precision float: sign bit, 11 bits exponent, 52 bits mantissa
      - 8
      - little-endian
-   * - `>f2`
+   * - ``>f2``
      - half precision float: sign bit, 5 bits exponent, 10 bits mantissa
      - 2
      - big-endian
-   * - `>f4`
+   * - ``>f4``
      - single precision float: sign bit, 8 bits exponent, 23 bits mantissa
      - 4
      - big-endian
-   * - `>f8`
+   * - ``>f8``
      - double precision float: sign bit, 11 bits exponent, 52 bits mantissa
      - 8
      - big-endian
+   * - ``r*`` (Optional)
+     - raw bits/bytes,  use for extension type fallbacks
+     - variable, given by ``*``, multiple of 8
```

Comment:
We'll discuss that today in the spec, my main questions was whether we want that to be the number of bits, or the number of bytes.

If it's number of Bytes, it can likely be any int.
The question is if we allow bits for types shorter the one byte (categorical or boolean), and define a behavior to pad/truncate.

Reply:
@QuLogic, it would be nice to get your thoughts on this (and other bit related points). Would like to make sure this makes sense for architectures that you are thinking about 🙂

Reply:
I think generally all systems in Fedora use 8-bit bytes, so there is no worry there. The key issue is order. I see there are definitions for endian order on multi-byte values, but I don't see anything about bit order (for a `packbits`-equivalent type, it looks like). If there's padding, is it between every entry, or just at the end? If the number of bits is > 8, is it first byte, then second, or the other way? Are the >8 bits in the lower or upper bits of the next byte? This is somewhat explained in examples, but not exactly explicit.

Reply:
Padding would just be at the end; not in between. I have no objections in changing. I think whether the order is lower first or lower last would be up to the extensions, or library using this which actually know the internal types. `r` is mostly meant as a pass through for when you can't fallback on an another of the type you can fallback on. Though I don't have that much experience with different endianness, so if you think this should be explicit I'll trust you.